standard/lexical-structure.md

```diff
@@ -324,64 +340,94 @@ Multiple translations are not performed. For instance, the string literal `"\u00
 The rules for identifiers given in this subclause correspond exactly to those recommended by the Unicode Standard Annex 15 except that underscore is allowed as an initial character (as is traditional in the C programming language), Unicode escape sequences are permitted in identifiers, and the "`@`" character is allowed as a prefix to enable keywords to be used as identifiers.
 
 ```ANTLR
-Identifier
+identifier
+    : Simple_Identifier
+    | contextual_keyword
+    ;
+
+Simple_Identifier
```

Comment:
You might wonder why *Simple_Identifier* exists and its simply not folded into *identifier*. So that the main lexical rules can be represented as a lexical grammar, rather than a combined grammar, in ANTLR the C# "lexical" rules are split into two ANTLR grammars (a) a lexical one for lexical rules and (b) a parser one for the parser rules. This arrangement will allow the pre-processor to be implemented as such an implementation is likely to rely on features only ANTLR lexical-only grammars can use. The rule `Simple_Identifer` provides the bridge between the lexer and parser rules of the "lexical" grammar... Now you know why I didn't explain that in the text, too tortuous! And its not as though the grammar doesn't have plenty of redundant rules for the purposes of exposition already, and the next one is immediately below...